### PR TITLE
[nav] SearchPoints LCM

### PR DESCRIPTION
**Pull request by zwgold**
Nav will send the search points when we are in the search turn and drive modes so that TeleOp can project them on the map when we perform our search spiral.

### CODE REVIEW

**Comment by zwgold** [comment]
Right now, the only states that send the paths are searchTurn and searchDrive. Should we send them when we go into obstacle mode or when we locate a target? Or even send them constantly and have TeleOp handle logic when it's empty or in certain states?

**Comment by zwgold** [comment]
Updated to always publish in searchStateMachine when those states within it are running

File: jetson/nav/search/searchStateMachine.cpp [hunk 1]
```diff
@@ -150,6 +151,20 @@ NavState SearchStateMachine::executeDriveToTarget() {
     return NavState::DriveToTarget;
 } // executeDriveToTarget()
 
+// Sends search path rover takes when trying to find posts
+void SearchStateMachine::publishSearchPoints() {
+    // Construct vector from deque
+    std::vector<Odometry> arr(mSearchPoints.begin(), mSearchPoints.end());
+    SearchPoints searchPoints{
+        .search_pattern_size  = (int32_t)mSearchPoints.size(),
```
Comment (cgiger00) [inline jetson/nav/search/searchStateMachine.cpp:159]: in general kind of a nitpick can you use ```std::static_cast<>```

Comment (zwgold) [reply jetson/nav/search/searchStateMachine.cpp:159]: yeah lemme fix that, good call

**Review by cgiger00 — APPROVED** [review]
lgtm

**Pull request merged**